default.json

```
@@ -7,7 +7,7 @@
       "customType": "regex",
       "fileMatch": [".*"],
       "matchStrings": [
-        "renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>\\S+?)(?: (packageName)=(?<packageName>\\S+?))?(?: versioning=(?<versioning>\\S+?))?(?: registryUrl=(?<registryUrl>\\S+?))?\\n\\s*(?:\\S+?_)?(?:VERSION|version|TAG|tag)\\s*[?]?[=:]\\s*\"?(?<currentValue>\\S+?)\"?\\s"
+        "renovate:\\s*datasource=(?<datasource>\\S+)\\s+depName=(?<depName>\\S+)(?:\\s+(package|lookup)Name=(?<packageName>\\S+))?(?:\\s+versioning=(?<versioning>\\S+))?(?:\\s+registryUrl=(?<registryUrl>\\S+))?\\n.+?_(?:VERSION|version|TAG|tag)\\s*[?=:]+\\s*\"?(?<currentValue>[^\" \\t\\n]+)\"?"

```

I dislike allowing variable spacing between the parts of the marker, I think it is nice that these are quite standardized, e.g. not allowing:

```yaml
renovate:datasource=some-datasource         depName=some-depname
packageName=some-package-name
```

I like that it is a single line containing a single space between each, making it consistent. This is also how the upstream regex-presets works: https://docs.renovatebot.com/presets-regexManagers/

---

`lookupName` is [deprecated](https://github.com/renovatebot/renovate/issues/14341), I don't think we should support that and change our own markers if we depend on that.

---

Upstream is datasource `[a-z-.]+?`, why change to `\\S+`?

---

You are making the matcher greedy instead of non-greedy, what is the rationale behind this? Upstream uses non-greedy.

---

> I dislike allowing variable spacing between the parts of the marker, I think it is nice that these are quite standardized

I think it is more important that it works.

> lookupName is deprecated, I don't think we should support that and change our own markers if we depend on that.

Agreed

> Upstream is datasource [a-z-.]+?, why change to \\S+?

Shorter? And a bit more flexible?

> You are making the matcher greedy instead of non-greedy, what is the rationale behind this? Upstream uses non-greedy.

@sverrehu You should answer this.

---

> I think it is more important that it works.

It works, but users can't put in tabs, newlines, and multiple spaces in the middle of their markers, I think it is a quite sane requirement to require a single space between each marker component.

---

I think we should just to as Mikael says. The regex was grabbed from a two years old config in one of our internal projects, and was not checked for consistency with upstream. Since it is now moved to the top-level config, consistency becomes more important.

---

```suggestion
        "renovate:\\s*datasource=(?<datasource>\\S+)\\s+depName=(?<depName>\\S+)(?:\\s+(package|lookup)Name=(?<packageName>\\S+))?(?:\\s+versioning=(?<versioning>\\S+))?(?:\\s+registryUrl=(?<registryUrl>\\S+))?\\n.+?_(?:VERSION|version|TAG|tag)\\s*[?=:]+\\s*\"?(?<currentValue>[^\"\\s]+)\"?"
```

I think escape sequences are allowed inside character classes.

---

The only important part in this change, is to allow optional double quotes around `currentValue`.

---

Double quotes around `currentValue` should be supported with the current: https://regex101.com/r/APv8b2/1

But I agree that the proposed change in the regex with regards to that is a better solution (maybe with the added suggestion of me to use the escape sequence for space characters).

---

Agree that parenthesis around `packageName` is no longer needed.